src/Control/Algebra.hs

```diff
@@ -52,6 +52,11 @@ class (HFunctor sig, Monad m) => Algebra sig m | m -> sig where
   -- | Construct a value in the carrier for an effect signature (typically a sum of a handled effect and any remaining effects).
   alg :: sig m a -> m a
 
+instance Algebra Pure PureC where
+  alg v = case v of {}
+  {-# INLINE alg #-}
```

Comment:
Defining the `Algebra` for `PureC` in here means we don’t need a `.hs-boot` file for it any more.